src/target/linux.rs

```diff
@@ -137,6 +165,31 @@ impl LinuxTarget {
             })
             .collect())
     }
+
+    /// Returns the current snapshot view of this debugee process threads.
+    pub fn threads(
+        &self,
+    ) -> Result<Vec<Box<dyn Thread<ThreadId = i32>>>, Box<dyn std::error::Error>> {
+        let tasks: Vec<_> = Process::new(self.pid.as_raw())?
+            .tasks()?
+            .flatten()
+            .collect();
+
+        let mut result: Vec<Box<dyn Thread<ThreadId = i32>>> = vec![];
+        for task in tasks {
+            match task.stat() {
+                Ok(t_stat) => {
+                    let thread = LinuxThread::new(&t_stat.comm, task.tid);
+                    result.push(Box::new(thread));
+                }
+                Err(ProcError::NotFound(_)) | Err(ProcError::Incomplete(_)) => {
+                    // ok to skip. Thread is gone or it's page is not complete yet.
+                }
+                Err(err) => return Err(Box::new(err)),
+            }
```

Comment:
A thread could change it's own name I think. Maybe LinuxThread should only store pid and tid and name() would read the current name or return None when the thread doesn't exist anymore?

Reply:
Ah... interesting. Ok will check this later today then. 

Regarding keeping the pid and tid: as far as I can tell they are the same thing.
So we only need to keep one. Is this correct?

Reply:
`procfs::process::Task` contains the pid of the whole process and the tid of the specific thread. You could also directly store that struct.

Reply:
You are right, a thread can change the name. From [ man procfs](https://man7.org/linux/man-pages//man5/procfs.5.html): 
> A thread may modify its comm value, or that of any of other thread in the same thread group 

Going with your idea of having the ```name``` method return the most recent: do you think it will be fine to swallow all other [procfs::ProcError](https://docs.rs/procfs/0.8.0/procfs/enum.ProcError.html) into a None? If so, the user wouldn't know something went wrong. Should we use a result on the name? 

Also I'm not sure I understand the benefits of the change. If you are getting a list of threads you either have them all paused or as soon as you get your information it might already be outdated. So the user will always have this issue of seeing old data, even if we push it elsewhere. He will always need to poll the thread list to get up-to-date information.

Reply:
> Should we use a result on the name?

Yes

> If you are getting a list of threads you either have them all paused or as soon as you get your information it might already be outdated.

I think once we support actually debugging threads, we would store a list of all active threads. When using `PTRACE_O_TRACECLONE` to automatically attach to new threads we have to manually continue the new thread anyway, so we could just as well store it in a list. Only when attaching to a new process do we need to list all active threads. Making the `LinuxThread` persistent also has the benefit that you could continue and pause individual threads rather than the whole process.

Reply:
Done. 

> I think once we support actually debugging threads, we would store a list of all active threads. When using PTRACE_O_TRACECLONE to automatically attach to new threads we have to manually continue the new thread anyway, so we could just as well store it in a list. Only when attaching to a new process do we need to list all active threads. Making the LinuxThread persistent also has the benefit that you could continue and pause individual threads rather than the whole process.

Alright, I completely missed that the first time you mentioned. I think I get it now - thank you!. We are going to be in lockstep with threads getting notified through ptrace that a new thread was created and adding that to the list of the process.